packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogLogsPlugin.kt

```diff
@@ -26,119 +26,150 @@ class DatadogLogsPlugin : MethodChannel.MethodCallHandler {
-                        result.success(null)
+            }
+            "addTag" -> {
+                val tag = call.argument<String>(LOG_TAG)
+                if (tag != null) {
+                    val value = call.argument<String>(LOG_VALUE)
+                    if (value != null) {
+                        logger.addTag(tag, value)
                     } else {
-                        result.missingParameter(call.method)
+                        logger.addTag(tag)
                     }
+                    result.success(null)
+                } else {
+                    result.missingParameter(call.method)
                 }
-                "removeTag" -> {
-                    val tag = call.argument<String>(LOG_TAG)
-                    if (tag != null) {
-                        log.removeTag(tag)
-                        result.success(null)
-                    } else {
-                        result.missingParameter(call.method)
-                    }
+            }
+            "removeAttribute" -> {
+                val key = call.argument<String>(LOG_KEY)
+                if (key != null) {
+                    logger.removeAttribute(key)
+                    result.success(null)
+                } else {
+                    result.missingParameter(call.method)
                 }
-                "removeTagWithKey" -> {
-                    val key = call.argument<String>(LOG_KEY)
-                    if (key != null) {
-                        log.removeTagsWithKey(key)
-                        result.success(null)
-                    } else {
-                        result.missingParameter(call.method)
-                    }
+            }
+            "removeTag" -> {
+                val tag = call.argument<String>(LOG_TAG)
+                if (tag != null) {
+                    logger.removeTag(tag)
+                    result.success(null)
+                } else {
+                    result.missingParameter(call.method)
                 }
-                else -> {
-                    result.notImplemented()
+            }
+            "removeTagWithKey" -> {
+                val key = call.argument<String>(LOG_KEY)
+                if (key != null) {
+                    logger.removeTagsWithKey(key)
+                    result.success(null)
+                } else {
+                    result.missingParameter(call.method)
                 }
             }
-        } catch (e: ClassCastException) {
-            result.error(
-                DatadogSdkPlugin.CONTRACT_VIOLATION, e.toString(),
-                mapOf(
-                    "methodName" to call.method
-                )
-            )
+            else -> {
+                result.notImplemented()
+            }
+        }
+    }
+
+    private fun createLogger(loggerHandle: String, configuration: LoggingConfiguration) {
+        val logBuilder = Logger.Builder()
+            .setDatadogLogsEnabled(true)
```

Comment:
shouldn't we read `configuration.sendLogsToDatadog` here? Although it is useful only for SDK development probably.

Reply:
Yes, good catch.